plugins/repository-s3/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -56,4 +56,10 @@ grant {
 
   // only for tests : org.opensearch.repositories.s3.S3RepositoryPlugin
   permission java.util.PropertyPermission "opensearch.allow_insecure_settings", "read,write";
+
+  permission java.util.PropertyPermission "aws.sharedCredentialsFile", "read,write";
+  permission java.util.PropertyPermission "aws.configFile", "read,write";
+  permission java.util.PropertyPermission "opensearch.path.conf", "read,write";
```

Comment:
Probably only read for `opensearch.path.conf`

Reply:
We need write permission also, since we are setting this property in tests to point to a config path, because otherwise the [override for shared credentials folder and config file](https://github.com/opensearch-project/OpenSearch/pull/7372/files?diff=split&w=1#diff-ada10ff196f861f6936364fc59dc9ca6f2669a9b6dfc1d23aecf1f73de9ce675R198) throws a `NullPointerException`.

Ref: https://github.com/opensearch-project/OpenSearch/pull/7372/files?diff=split&w=1#diff-dd07f9badd52bf2005d3f466d4112b7f137a8dd7849388ba5060e7d0a3cb6f7dR62

Reply:
For tests, we should be using test policy, this is the policy we are deploying on production clusters, no settings needed by tests should be there

Reply:
We've changed it for both plugins and reduced the policy to a minimum to pass tests.